taxi/forms.py

```diff
@@ -0,0 +1,66 @@
+import re
+
+from django import forms
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+
+from taxi.models import Driver, Car
+
+
+def validate_license_number(license_number):
+    if len(license_number) != 8:
+        raise ValidationError("License number length must be 8")
+
+    if not re.match("[A-Z]{3}", license_number[:3]):
+        raise ValidationError(
+            "First 3 characters must be uppercase letters"
+        )
+
+    if not license_number[3:].isdigit():
+        raise ValidationError("Last 5 characters must be digits")
+
+    return license_number
+
+
+class DriverLicenseUpdateForm(forms.ModelForm):
+    class Meta:
+        model = Driver
+        fields = (
+            "last_name",
+            "first_name",
+            "license_number",
+        )
+
+    def clean_license_number(self):
+        return validate_license_number(self.cleaned_data.get("license_number"))
+
+
+class DriverLicenseCreateForm(forms.ModelForm):
+    cars = forms.ModelMultipleChoiceField(
```

Comment:
Whats the point to include cars here? These relations will not be saved in generic CreateView because it's a backwards relationship - you will need to add logic in view to save m2m relations then